add tidb-3.0 compatibility document (#998)
pingcap/tispark: README.md

## Statistics information
If you want to know how TiSpark could benefit from TiDB's statistic information, read more [here](./docs/userguide.md).

## Compatibility with tidb-3.0
### View

> TiDB start to support `view` from `tidb-3.0`. 

Reviewer: s/start/starts

> TiSpark currently does not support `view` and will filter all views from catalog. Users cannot select data from views through TiSpark.

Reviewer 2: ```suggestion
TiSpark currently **does not support** `view`. Users will not be able to observe or access data through views with TiSpark.
```

### Table Partition
`tidb-3.0` supports both `Range Partition` and `Hash Partition`. 

> TiSpark currently does support `Range Partition` and `Hash Partition`. Users can select data from `Range Partition` table and `Hash Partition` table through TiSpark. 

Reviewer 2: ```suggestion
TiSpark currently **support** `Range Partition` and `Hash Partition`. Users can select data from `Range Partition` table and `Hash Partition` table through TiSpark. 
```

> In most cases TiSpark will use full table scan. Only in such cases TiSpark will apply partition pruning:

Reviewer: s/Only in such cases TiSpark will apply partition pruning/Only in the following cases TiSpark will apply partition pruning to build a better plan

- table is `Range Partition`

> - partition expression only contains function `year`.

Reviewer: partition expression is column or year expression whose child is column.

> - partition expression is a `column` expression or `year` expression whose child is a column expression

Reviewer: and its type is datetime or string literal but can be parsed as datetime.
